### PR TITLE
Make the default configuration cluster ready

### PR DESCRIPTION
**Pull request by cimnine**
This changes the default configuration to be better prepared for
usage with container management platforms, such as Docker Swarm,
Kubernetes or OpenShift.

Inspired by #27.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,13 +56,64 @@ To ensure this, compare the output of `docker --version` and `docker-compose --v
...
+### Production
+
+The default settings are optimized for (local) development environments.
+You should therefore adjust the configuration for production setups, at least the following variables:
+
+* `ALLOWED_HOSTS`: Add all URLs that lead to your netbox instance.
+* `DB_*`: Use a persistent database.
+* `EMAIL_*`: Use your own mailserver.
+* `MAX_PAGE_SIZE`: Use the recommended default of 1000.
+* `SUBERUSER_*`: Only define those variables during the initial setup, and drop them once the DB is set up.
```
Comment (twhiston) [inline README.md]: is SUBERUSER deliberate? or should it be superuser?

File: README.md [hunk 2]
```diff
@@ -51,13 +56,64 @@ To ensure this, compare the output of `docker --version` and `docker-compose --v
...
+### Running on Docker Swarm / Kubernetes / OpenShift
+
+You may run this image in a cluster such as Docker Swarm, Kubernetes or OpenShift, but this is advanced level.
+
+In this case, we encourage you to statically configure Netbox by starting from [Netbox's example config file][default-config], and mounting it into your container using the mechanism provided by your container platform (i.e. [Docker Swarm configs][swarm-config], [Kubernetes secrets][k8s-secrets], [OpenShift configmaps][openshift-config]).
+
+But if you rather continue to configure your application through environment variables, you may continue to use [the built-in configuration file][docker-config].
+We discourage storing secrets in environment variables, as environment variable are passed on to all sub-processes and may leak easily into other systems, e.g. error collecting tools that often collect all environment variables whenever an error occurs.
+
+Therefore we *strongly advise* to make use of the secrets mechanism provided by your container platform (i.e. [Docker Swarm secrets][swarm-secrets], [Kubernetes secrets][k8s-secrets], [OpenShift secrets][openshift-secrets]).
```
Comment (twhiston) [inline README.md]: ^ this is all good advice :D like it

File: docker-compose.yml [hunk 3]
```diff
@@ -31,3 +34,7 @@ volumes:
         driver: local
     netbox-nginx-config:
         driver: local
+    netbox-media-files:
```
Comment (twhiston) [inline docker-compose.yml]: i cant comment on the line above, as it didnt change, but do we really want to hardcode the nginx port to 80, as that barely ever binds correctly (osx default apache or whatever)

File: README.md (hunk 1 above)
Comment (cimnine) [reply README.md]: It's deliberate, as those are environment variables and therefore capital by convention.

File: docker-compose.yml (hunk 3 above)
Comment (cimnine) [reply docker-compose.yml]: That's only the 'container internal port'. `docker-compose` will bind that port to a randomish port.

File: README.md (hunk 1 above)
Comment (twhiston) [reply README.md]: not capitalization, spelling, suBeruser

**Comment by cimnine** [comment]
Good thing that there are (some) unit tests 😇

**Pull request merged**